python/hsml/constants.py

```diff
@@ -38,11 +38,12 @@ class ARTIFACT_VERSION:
 
 
 class RESOURCES:
+    MIN_NUM_INSTANCES = 1  # disable scale-to-zero by default
```

Comment:
For app we do want the scale to zero enabled though if I remember correctly, is that being handled somewhere?

Reply:
Yes, we check if scale-to-zero is required when getting the default resources [here](https://github.com/javierdlrm/machine-learning-api/blob/6e6f058875267d4e8d721c64aab3caead3bbc95b/python/hsml/predictor.py#L201) and [here](https://github.com/javierdlrm/machine-learning-api/blob/6e6f058875267d4e8d721c64aab3caead3bbc95b/python/hsml/transformer.py#L62).

While supporting non-KServe deployments, we cannot default to scale-to-zero (e.g., num_instances = 0).